From 99967395f97cd45b0dbb4255a56305a8bd62c5c1 Mon Sep 17 00:00:00 2001
From: Omair Majid <omajid@redhat.com>
Date: Wed, 1 Nov 2017 10:50:17 -0400
Subject: [PATCH] Dont strip native debug symbols during build

This patch makes coreclr and core-setup leave debug symbols in their
built native code instead of stripping it out.

After this patch, native symbols are only missing in corefx. A separate
PR in corefx will take care of that:
https://github.com/dotnet/corefx/pull/24979

The final dotnet product, after both these fixes, should contain native
debug symbols for all native binaries.
---
 repos/core-setup.proj | 2 +-
 repos/coreclr.proj    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/repos/core-setup.proj b/repos/core-setup.proj
index 4890ac5..65b67c0 100644
--- a/repos/core-setup.proj
+++ b/repos/core-setup.proj
@@ -2,7 +2,7 @@
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))/dir.props" />
   <PropertyGroup>
-    <BuildArguments>-ConfigurationGroup=$(Configuration) -PortableBuild=false -strip-symbols -SkipTests=true </BuildArguments>
+    <BuildArguments>-ConfigurationGroup=$(Configuration) -PortableBuild=false -SkipTests=true </BuildArguments>
     <BuildArguments Condition="$(Platform.Contains('arm'))">$(BuildArguments) -TargetArchitecture=$(Platform) -DisableCrossgen=true -CrossBuild=true</BuildArguments>
     <BuildCommand>$(ProjectDirectory)/build$(ShellExtension) $(BuildArguments) -- /p:BuildDebPackage=false</BuildCommand>
     <BuildCommand Condition="$(Platform.Contains('arm'))">$(ArmEnvironmentVariables) $(BuildCommand)</BuildCommand>
diff --git a/repos/coreclr.proj b/repos/coreclr.proj
index f1e4de9..4078d73 100644
--- a/repos/coreclr.proj
+++ b/repos/coreclr.proj
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <BuildArguments>$(Platform) $(Configuration) skiptests -PortableBuild=false </BuildArguments>
     <BuildArguments Condition="'$(OS)' != 'Windows_NT'">$(BuildArguments) msbuildonunsupportedplatform</BuildArguments>
-    <BuildArguments Condition="$(Platform.Contains('arm'))">$(BuildArguments) skipnuget cross -skiprestore stripSymbols cmakeargs -DFEATURE_GDBJIT=TRUE</BuildArguments>
+    <BuildArguments Condition="$(Platform.Contains('arm'))">$(BuildArguments) skipnuget cross -skiprestore cmakeargs -DFEATURE_GDBJIT=TRUE</BuildArguments>
     <BuildCommand>$(ProjectDirectory)/build$(ShellExtension) $(BuildArguments) --</BuildCommand>
     <BuildCommand Condition="$(Platform.Contains('arm'))">$(ArmEnvironmentVariables) $(BuildCommand)</BuildCommand>
     <CleanCommand>$(ProjectDirectory)/clean$(ShellExtension)</CleanCommand>
-- 
2.7.4

